pyqtgraph/graphicsItems/ScatterPlotItem.py

```diff
@@ -164,6 +171,29 @@ def _mkBrush(*args, **kwargs):
         return fn.mkBrush(*args, **kwargs)
 
 
+class PixmapFragments:
+    def __init__(self):
+        self._array = np.empty((0, 10), dtype=np.float64)
+        self.ptrs = None
+
+    def array(self, size):
+        if size > self._array.shape[0]:
+            self._array = np.empty((size + 16, 10), dtype=np.float64)
+            if QT_LIB.startswith('PyQt'):
+                self.ptrs = list(map(sip.wrapinstance,
+                    itertools.count(self._array.ctypes.data, self._array.strides[0]),
+                    itertools.repeat(QtGui.QPainter.PixmapFragment, self._array.shape[0])))
+            else:
+                self.ptrs = wrapInstance(self._array.ctypes.data, QtGui.QPainter.PixmapFragment)
+        return self._array[:size]
+
+    def draw(self, painter, size, pixmap):
+        if QT_LIB.startswith('PyQt'):
+            painter.drawPixmapFragments(self.ptrs[:size], pixmap)
```

Comment:
This throws an error for zero length data: self.ptrs is None in that case.

Reply:
> If you look at the python bindings, `drawLines`, `drawRects`, `drawPoints` do not expose their pointer to structs api, whereas `drawPixmapFragments` exposes it.

> So, in fact, rather than add numpy specific apis like `drawPointsNp`, it would be better to just expose the pointer to structs c++ api, which would be numpy-independent. After all, QRectF, QLineF, QPointF are just a struct of 4,4,2 float64s respectively in memory, Though, of course, it is non-Pythonic and non-type-safe.

@j9ac9k have you been in touch with the PySide developers about this?

Reply:
I opened an issue regarding `drawPointsNp` only got a brief reply but suggested they were leaning against implementing it as they didn't think it was general enough.

I did ask on their gitter channel if it would be feasible to expose the pointer for the underlying struct of data the other draw methods take.  Hard for me to gauge work effort involved there.

Reply:
fixed